src/thor/map_matcher.cc

```diff
@@ -177,8 +174,8 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
 
   // We support either the epoch timestamp that came with the trace point or
   // a local date time which we convert to epoch by finding the first timezone
-  uint32_t origin_epoch = compute_origin_epoch(edge_segments, matcher, options);
-  std::string date_time = options.shape(0).has_date_time() ? options.shape(0).date_time() : "";
+  valhalla::baldr::DateTime::tz_sys_info_cache_t tz_cache;
+  auto time_info = init_time_info(edge_segments, matcher, options, &tz_cache);
```

Comment:
this is the main point of this pr. we dont manually track second of week or the timezone etc. we do it with the time_info object as all the other time based algorithms do it